management/src/main/java/org/kaazing/gateway/management/monitoring/entity/manager/impl/ServiceSessionCounterManagerImpl.java

```diff
@@ -44,26 +45,28 @@
 
     private MonitoringEntityFactory monitoringEntityFactory;
     private String serviceName;
+    private String gatewayId;
 
     public ServiceSessionCounterManagerImpl(MonitoringEntityFactory monitoringEntityFactory,
-            String serviceName) {
+            String serviceName, String gatewayId) {
         this.monitoringEntityFactory = monitoringEntityFactory;
         this.serviceName = serviceName;
+        this.gatewayId = gatewayId;
    }
 
     @Override
     public void initializeCounters() {
-        numberOfSessionsCounter = monitoringEntityFactory.makeLongMonitoringCounter(serviceName
+        numberOfSessionsCounter = monitoringEntityFactory.makeLongMonitoringCounter(gatewayId + SEPARATOR + serviceName
```

Comment:
Is it possible to get rid of the LongMonitoringCounter abstraction and just use Agrona's AtomicCounter (as in Aeron's SystemCounters class)? Not a big deal, but I'm curious to know why we need that extra abstraction.